### DISCUSSION TITLE
ImageMagick on Debian-Buster without Delegates

### DISCUSSION THREAD

**Discussion by DaWiWhite** [ImageMagick/ImageMagick#2634] [Help] [answered]
My first Discussion - Entry -> so i try to give my best :-) 

Hello to everyone, 

i installed a bare DEBIAN Buster from scratch. I dont enable any UI because i use it only for server purposes. 
My intention is to become a private MEDIAWiki - Server for my informations :-)

Now i would have the possibility to have images and preview-images in MediaWiki so i installed ImageMagick according to the instructions here:

```
$ sudo apt update
$ sudo apt-get install build-essential
$ wget https://www.imagemagick.org/download/ImageMagick.tar.gz
$ tar xvzf ImageMagick.tar.gz
$ cd ImageMagick-7.0.8-26/         -- i used the actually version
$./configure 
$ make
$ sudo make install 
$ sudo ldconfig /usr/local/lib
$ magick -version
```

i used the instructions from https://www.tecmint.com/install-imagemagick-on-debian-ubuntu/

But if i test the installed **Delegates** with

`$ magick -version `

There are none... i need PNG and JPG and so on... the usual...

So i installed **libpng** first with `sudo apt-get install libpng-dev` and do the instructions from above one more time....

but no improvement...

can someone help me please? 

THX

**Comment by DaWiWhite** [comment, marked as answer]
Obviously this is because the filters for the respective **image or file formats were not installed** in the operating system (DEBIAN). 

I have found a list of file format filters to install which you have to install in Debian. 

```
sudo apt-get install autoconf automake autopoint autotools-dev build-essential chrpath cm-super-minimal debhelper dh-autoreconf dh-exec dh-strip-nondeterminism doxygen doxygen-latex dpkg-dev fonts-lmodern g++ g++-7 gcc gcc-7 gir1.2-harfbuzz-0.0 graphviz icu-devtools libann0 libasan4 libatomic1 libbz2-dev libc-dev-bin libc6-dev libcairo-script-interpreter2 libcairo2-dev libcdt5 libcgraph6 libcilkrts5 libclang1-6.0 libdjvulibre-dev libexif-dev libexpat1-dev libfftw3-bin libfftw3-dev libfftw3-long3 libfftw3-quad3 libfile-stripnondeterminism-perl libfontconfig1-dev libfreetype6-dev libgcc-7-dev libgdk-pixbuf2.0-dev libglib2.0-dev  libglib2.0-dev-bin libgraphite2-dev libgts-0.7-5 libgvc6 libgvpr2 libharfbuzz-dev libharfbuzz-gobject0 libice-dev libicu-dev libicu-le-hb-dev libicu-le-hb0 libilmbase-dev libitm1 libjbig-dev libjpeg-dev libjpeg62-turbo-dev liblab-gamut1 liblcms2-dev liblqr-1-0-dev liblsan0 libltdl-dev liblzma-dev libmime-charset-perl libmpx2 libopenexr-dev libpango1.0-dev libpathplan4 libpcre16-3 libpcre3-dev libpcre32-3 libpcrecpp0v5 libperl-dev libpixman-1-dev libpng-dev libpotrace0 libptexenc1 libpthread-stubs0-dev libpython-stdlib libquadmath0 librsvg2-bin librsvg2-dev libsigsegv2 libsm-dev libsombok3 libstdc++-7-dev libtexlua52 libtexluajit2 libtiff-dev libtiff5-dev libtiffxx5 libtool libtool-bin libtsan0 libubsan0 libunicode-linebreak-perl libwmf-dev libx11-dev libxau-dev libxcb-render0-dev libxcb-shm0-dev libxcb1-dev libxdmcp-dev libxext-dev libxft-dev libxml2-dev libxml2-utils libxrender-dev libxt-dev libzzip-0-13 linux-libc-dev m4 make pkg-config pkg-kde-tools po-debconf preview-latex-style python python-minimal python2.7 python2.7-minimal python3-distutils python3-lib2to3 tex-common texlive-base texlive-binaries texlive-extra-utils texlive-font-utils texlive-fonts-recommended texlive-latex-base texlive-latex-extra texlive-latex-recommended texlive-pictures x11proto-core-dev x11proto-dev x11proto-xext-dev xorg-sgml-doctools xsltproc xtrans-dev zlib1g-dev
```

After that just install ImageMagick again and the corresponding delegates are used... `magick -version`

**Reply by dlemstra** [reply]
We are a very small team and we don't have a lot of spare time to work on this project. So yes it can take a while before we have time to research your question and respond to it. And instead of being "funny" you could also help the next person who runs into the same issue. It would be very helpful to them if you would explain how you solved this.

**Comment by DaWiWhite** [comment]
Hey dlemstra, 

I understand your argumentation and have edited my answer so that others don't have to die stupidly either...

Don't be angry, but I really felt very ALONE...
